### PR TITLE
Add automation for Netperf with Azure VMs 

### PR DESCRIPTION
**Pull request by ProjectsByJackHe**
## Description

Schema for Sqlite database:

**-- Table: Environment**
CREATE TABLE IF NOT EXISTS Environment (
        Environment_ID INTEGER PRIMARY KEY AUTOINCREMENT,
        OS             TEXT,
        OS_type        TEXT,
        OS_version     TEXT,
        Architecture   TEXT,
        [NIC_type ]    TEXT,
        CPU_type       TEXT
);

**-- Table: Secnetperf_builds**
CREATE TABLE IF NOT EXISTS Secnetperf_builds (
    Secnetperf_build_ID   INTEGER PRIMARY KEY AUTOINCREMENT,
    Secnetperf_commit     TEXT,
    Build_date_time       TEXT,
    TLS_enabled           INTEGER,
    Advanced_build_config TEXT
);

**-- Table: Secnetperf_latency_stats**
CREATE TABLE IF NOT EXISTS Secnetperf_latency_stats (
    Secnetperf_latency_stats_ID INTEGER PRIMARY KEY AUTOINCREMENT,
    P0                          REAL,
    P50                         REAL,
    P90                         REAL,
    P99                         REAL,
    P999                        REAL,
    P9999                       REAL,
    P99999                      REAL,
    P999999                     REAL
);

**-- Table: Secnetperf_test_runs**
CREATE TABLE IF NOT EXISTS Secnetperf_test_runs (
    Secnetperf_test_ID          INTEGER REFERENCES Secnetperf_tests (Secnetperf_test_ID),
    Client_environment_ID       INTEGER REFERENCES Environment (Environment_ID),
    Server_environment_ID       INTEGER REFERENCES Environment (Environment_ID),
    Result                      REAL,
    Secnetperf_latency_stats_ID INTEGER
);

**-- Table: Secnetperf_tests**
CREATE TABLE IF NOT EXISTS Secnetperf_tests (
    Secnetperf_test_ID INTEGER PRIMARY KEY AUTOINCREMENT,
    Kernel_mode        INTEGER,
    Run_arguments      TEXT
);

## Testing

CI & RDP validation

## Documentation

Docs added in netperf/docs

### CODE REVIEW

File: scripts/secnetperf.ps1 [hunk 1]
```diff
@@ -5,8 +5,17 @@ NOTE:
...
 
+.PARAMETER Logging
+    If true, will enable logging for the test. This will create a log file in the artifacts directory.
+    Default is false.
+
 #>
 
+param (
+    [Parameter(Mandatory = $false, ParameterSetName='Logging')]
+    [switch]$Logging = $false,
```
Comment (nibanks) [inline scripts/secnetperf.ps1:16]: It should be a `String` (default to empty) to specify the logging profile instead.
```suggestion
    [ValidateSet("", "Basic.Light", "Datapath.Light", "Datapath.Verbose", "Stacks.Light", "Stacks.Verbose", "RPS.Light", "RPS.Verbose", "Performance.Light", "Basic.Verbose", "Performance.Light", "Performance.Verbose", "Full.Light", "Full.Verbose", "SpinQuic.Light", "SpinQuicWarnings.Light")]
    [string]$LogProfile = "",
```

File: scripts/secnetperf.ps1 [hunk 2]
```diff
@@ -85,17 +113,19 @@ Write-Output Wait-ForRemote $Job
...
-dir C:\Windows\system32\Logfiles\WMI\prof*
-Write-Output "Stopping session..."
-logman stop profsvc -ets -ErrorAction Ignore
-Start-Sleep 5
-Write-Output "Copying profsvc logs..."
-Copy-Item $env:WINDIR\System32\LogFiles\WMI\profsvc.etl.* .\artifacts\logs
-dir .\artifacts\logs
-#netsh trace convert .\artifacts\logs\profsvc.etl
-
+    if ($Logging) {
```
Comment (nibanks) [inline scripts/secnetperf.ps1:116]: We should only need to grab our logs. All these were for temporary testing.

Comment (ProjectsByJackHe) [reply scripts/secnetperf.ps1:116]: Noted.

File: scripts/secnetperf.ps1 [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+    $Job = Invoke-Command -Session $Session -ScriptBlock {
+        C:\_work\quic\artifacts\bin\windows\x64_Release_schannel\secnetperf.exe -exec:maxtput
+    } -AsJob
+} else {
+    $Job = Invoke-Command -Session $Session -ScriptBlock {
+        $env:LD_LIBRARY_PATH = "${env:LD_LIBRARY_PATH}:/home/secnetperf/_work/artifacts/bin/linux/x64_Release_openssl/"
+        chmod +x /home/secnetperf/_work/artifacts/bin/linux/x64_Release_openssl/secnetperf
+        /home/secnetperf/_work/artifacts/bin/linux/x64_Release_openssl/secnetperf -exec:maxtput
+    } -AsJob
+}
```
Comment (nibanks) [inline scripts/secnetperf.ps1:131]: These paths need to not be hard coded, but instead take the input parameters into account to build up the path.

File: scripts/secnetperf.ps1 [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+
+# Save as a .sql file
+$FileName = "test-results-$plat-$os-$arch-$tls.sql"
+
+Set-Content -Path $FileName -Value $SQL
+
+# Save as a .json file
+$FileName = "json-test-results-$plat-$os-$arch-$tls.json"
+
+$json | ConvertTo-Json | Set-Content -Path $FileName
```
Comment (nibanks) [inline scripts/secnetperf.ps1:293]: The comments in this section are a bit too verbose, and unnecessary.
```suggestion
# Save the test results (SQL & JSON).
Write-Output "Saving test results..."
Set-Content -Path "test-results-$plat-$os-$arch-$tls.sql" -Value $SQL
$json | ConvertTo-Json | Set-Content -Path "json-test-results-$plat-$os-$arch-$tls.json"
```

File: scripts/secnetperf.ps1 [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+
+} catch {
+    Write-Error "Failed to run some tests..."
+    Write-Error $_
+} finally {
+    # TODO: Do any further book keeping here.
```
Comment (nibanks) [inline scripts/secnetperf.ps1:299]: Is anything actually necessary here?

File: scripts/secnetperf.ps1 [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+    "throughput-upload-tcp-$MsQuicCommit",
+    "throughput-download-quic-$MsQuicCommit",
+    "throughput-download-tcp-$MsQuicCommit",
+    "rps-1conn-1stream-quic-$MsQuicCommit",
+    "rps-1conn-1stream-tcp-$MsQuicCommit",
+    "hps-quic-$MsQuicCommit",
+    "hps-tcp-$MsQuicCommit"
+)
+
+$commands = @(
```
Comment (nibanks) [inline scripts/secnetperf.ps1:191]: I'd like to remove the `-tcp:` option having to be explicitly specified in the list of commands. Every command can be run in QUIC or TCP mode. Let's remove it from the list below and dynamically add it in the loop below.

**Review by nibanks — APPROVED** [review]

**Pull request merged**